src/main/java/org/jitsi/videobridge/BitrateController.java

```diff
@@ -0,0 +1,601 @@
+    {
+        String endpointID = source.getEndpoint().getID();
+        return forwardedEndpoints.contains(endpointID);
+    }
+
+    /**
+     * The {@link PacketTransformer} that handles incoming/outgoing RTP
+     * packets for this {@link BitrateController} instance. Internally,
+     * it delegates this responsibility to the appropriate media stream track
+     * bitrate controller ({@link SimulcastController}, etc).
+     */
+    class RTPTransformer
+        implements PacketTransformer
+    {
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void close()
+        {
+            // TODO decrease counters.
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket[] reverseTransform(RawPacket[] pkts)
+        {
+            return pkts;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket[] transform(RawPacket[] pkts)
+        {
+            if (ArrayUtils.isNullOrEmpty(pkts))
+            {
+                return pkts;
+            }
+
+            for (int i = 0; i < pkts.length; i++)
+            {
+                if (pkts[i] == null
+                    || !RTPPacketPredicate.INSTANCE.test(pkts[i]))
+                {
+                    continue;
+                }
+
+                int ssrc = pkts[i].getSSRC();
+                SimulcastController subCtrl = subCtrls.get(ssrc);
+
+                if (subCtrl == null)
+                {
+                    pkts[i] = null;
+                    continue;
+                }
+
+                RawPacket[] transformedPkts = subCtrl
+                    .getRTPTransformer().transform(new RawPacket[]{pkts[i]});
+
+                pkts[i] = transformedPkts[0];
+            }
+
+            return pkts;
+        }
+    }
+
+    /**
+     * The {@link PacketTransformer} that handles incoming/outgoing RTCP
+     * packets for this {@link BitrateController} instance. Internally,
+     * it delegates this responsibility to the appropriate media stream track
+     * bitrate controller ({@link SimulcastController}, etc).
+     */
+    class RTCPTransformer
+        implements PacketTransformer
+    {
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void close()
+        {
+
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket[] reverseTransform(RawPacket[] pkts)
+        {
+            return pkts;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket[] transform(RawPacket[] pkts)
+        {
+            if (ArrayUtils.isNullOrEmpty(pkts))
+            {
+                return pkts;
+            }
+
+            for (int i = 0; i < pkts.length; i++)
+            {
+                if (pkts[i] == null
+                    || !RTPPacketPredicate.INSTANCE.test(pkts[i]))
```

Comment:
Shouldn't this be RTCP instead of RTP? And same comment as above for reusing SinglePacketTransforme.